Task 2/Task2.3/Trainings/Training.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace Task2._3.Trainings
+{
+    public class Training : TrainingEntity
+    {
+        private int _trainingsCount;
+        private BaseTraining[] _trainings;
+
+        public Training(string description)
+        {
+            Description = description;
+            _trainings = new BaseTraining[1];
+        }
+
+        public void Add(BaseTraining training)
+        {
+            if (_trainings.Length == _trainingsCount)
+            {
+                ResizeArray();
+            }
+            _trainings[_trainingsCount] = training;
+            _trainingsCount++;
+        }
+
+        public bool IsPractical()
+        {
+            foreach (var training in _trainings)
+            {
+                if (training is Lecture)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        public Training Clone()
+        {
+            var cloned = new Training(Description);
+
+            foreach (var training in _trainings)
```

Comment:
if-else and casting are not needed, just cloned.Add((BaseTraining)training.Clone());

Reply:
Fixed